integration/docker/kill_test.go

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package docker
+
+import (
+	"fmt"
+	"strings"
+	"syscall"
+	"time"
+
+	. "github.com/kata-containers/tests"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	canBeTrapped    = true
+	cannotBeTrapped = false
+)
+
+func withSignal(signal syscall.Signal, trap bool) TableEntry {
+	expectedExitCode := int(signal)
+	if !trap {
+		// 128 -> command interrupted by a signal
+		// http://www.tldp.org/LDP/abs/html/exitcodes.html
+		expectedExitCode += 128
+	}
+
+	return Entry(fmt.Sprintf("with '%d'(%s) signal", signal, syscall.Signal(signal)), signal, expectedExitCode, true)
+}
+
+func withoutSignal() TableEntry {
+	// 137 = 128(command interrupted by a signal) + 9(SIGKILL)
+	return Entry(fmt.Sprintf("without a signal"), syscall.Signal(0), 137, true)
+}
+
+func withSignalNotExitCode(signal syscall.Signal) TableEntry {
+	return Entry(fmt.Sprintf("with '%d' (%s) signal, don't change the exit code", signal, signal), signal, 0, false)
+}
+
+var _ = Describe("docker kill", func() {
+	var (
+		args []string
+		id   string
+	)
+
+	BeforeEach(func() {
+		id = randomDockerName()
+	})
+
+	AfterEach(func() {
+		Expect(RemoveDockerContainer(id)).To(BeTrue())
+		Expect(ExistDockerContainer(id)).NotTo(BeTrue())
+	})
+
+	DescribeTable("killing container",
+		func(signal syscall.Signal, expectedExitCode int, waitForExit bool) {
+			args = []string{"--name", id, "-dt", Image, "sh", "-c"}
+
+			switch signal {
+			case syscall.SIGQUIT, syscall.SIGILL, syscall.SIGBUS, syscall.SIGFPE, syscall.SIGSEGV, syscall.SIGPIPE:
+				Skip("This is not forwarded by cc-shim " +
```

Comment:
s/cc-shim/kata-shim/

Reply:
and move issues 769 and 768 to kata-container/shim